docs/se/guides/07_db_client.adoc

```diff
@@ -0,0 +1,604 @@
+    <artifactId>helidon-dbclient-metrics</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.dbclient</groupId>      <6>
+    <artifactId>helidon-dbclient-jsonp</artifactId>
+</dependency>
+----
+<1> DB Client API dependency.
+<2> Using JDBC driver for this example.
+<3> H2 driver dependency.
+<4> Support for health check.
+<5> Support for metrics.
+<6> Support for Jsonp.
+
+=== Configure the DB Client
+
+To configure the application, helidon uses by default the application.yaml. The DB client configuration can be joined
+in the same file. It is located here: `src/main/resources`.
+
+[source,yaml]
+.Copy those properties into application.yaml
+----
+db:
+  source: jdbc  # <1>
+  connection:   # <2>
+    url: "jdbc:h2:tcp://localhost:9092/~/test"
+    username: "sa"
+    password:
+  statements:   # <3>
+    #Ping statement required for health check
+    ping: "SET @HELIDON_PING=0"   # <4>
+    create-table: "CREATE TABLE IF NOT EXISTS LIBRARY (NAME VARCHAR NOT NULL, INFO VARCHAR NOT NULL)"
+    insert-book: "INSERT INTO LIBRARY (NAME, INFO) VALUES (:name, :info)"
+    select-book: "SELECT INFO FROM LIBRARY WHERE NAME = ?"
+    delete-book: "DELETE FROM LIBRARY WHERE NAME = ?"
+----
+<1> Source property support two values: jdbc and mongo.
+<2> Connection detail we used to set up H2.
+<3> SQL statements to manage the database.
+<4> The ping statement is mandatory because of health check.
+
+=== Build and set up Helidon DB Client
+
+In the application Main.class, an instance of DbClient is created based on the configuration from
+application.yaml.
+
+[source,java]
+.Create a DbClient in the Main.startServer method:
+----
+import io.helidon.dbclient.metrics.DbClientMetrics; // <1>
+import io.helidon.dbclient.DbClient;
+
+...
+
+Config config = Config.create(); // Landmark to add DB client
+
+DbClient dbClient = DbClient.builder()
+                .config(config.get("db"))       // <2>
+                .addService(DbClientMetrics.counter().statementNames("ping"))   // <3>
+                .build();
+----
+<1> Add import statements
+<2> Configure the DB Client with the "db" section of application.yaml.
+<3> Add a counter for metrics.
+
+The DB Client metric counter will be executed only for ping statement, and check how many times it was invoked.
+At this point, the database is empty, and needs to be initialised. To achieve that, the DB Client can be used
+to create a table in the database.
+
+[source,java]
+.Insert a createTable method below the dbClient:
+----
+DbClient dbClient = DbClient.builder()
+                .config(config.get("db"))
+                .addService(DbClientMetrics.counter().statementNames("ping"))
+                .build();
+
+createTable(dbClient);
+----
+
+[source,java]
+.Use the DbClient to build a table:
+----
+private static void createTable(DbClient dbClient) {
+        dbClient.execute(exec -> exec.namedDml("create-table"))     // <1>
+                .await();
```

Comment:
Please add a comment here - we can use `await()`, as this is part of bootstrapping of the server, and it does not make sense to start the server before the table is created.
In general usage of blocking operation in WebServer must be clearly explained.